pkg/build/admission/admission.go

```diff
@@ -107,6 +107,24 @@ func (a *buildByStrategy) checkBuildConfigAuthorization(buildConfig *buildapi.Bu
 	return a.checkAccess(strategyType, subjectAccessReview, attr)
 }
 
+func (a *buildByStrategy) checkBuildRequestAuthorization(req *buildapi.BuildRequest, attr admission.Attributes) error {
+	switch attr.GetResource() {
+	case buildsResource:
+		build, err := a.client.Builds(attr.GetNamespace()).Get(req.Name)
+		if err != nil {
+			return err
+		}
+		return a.checkBuildAuthorization(build, attr)
+	case buildConfigsResource:
+		build, err := a.client.BuildConfigs(attr.GetNamespace()).Get(req.Name)
+		if err != nil {
+			return err
+		}
+		return a.checkBuildConfigAuthorization(build, attr)
+	}
```

Comment:
if we get a build request from a type we don't recognize, I think we should error... it probably means something new got created that we need to handle here (or at least explicitly not handle)